From 74a1b363eab695d08bf302425a4ac5c338481440 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Mon, 17 Feb 2025 10:40:11 +0100
Subject: [PATCH] mangoapp: Hide the window when paused instead of iconifying.

Workaround a GLFW bug with Mutter and other WM that do not unmap windows
when iconified. A later glfwRestoreWindow which only calls XMapWindow is
no-op in that case and won't actually restore the window.

Link: https://github.com/glfw/glfw/issues/2077
---
 src/app/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/app/main.cpp b/src/app/main.cpp
index 4d1d3b1277..0e86a21021 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -363,7 +363,7 @@ int main(int, char**)
 
         if (!params.no_display){
             if (mangoapp_paused){
-                glfwRestoreWindow(window);
+                glfwShowWindow(window);
                 uint32_t value = 1;
                 XChangeProperty(x11_display, x11_window, overlay_atom, XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&value, 1);
                 XSync(x11_display, 0);
@@ -404,7 +404,7 @@ int main(int, char**)
 
             glfwSwapBuffers(window);
         } else if (!mangoapp_paused) {
-            glfwIconifyWindow(window);
+            glfwHideWindow(window);
             uint32_t value = 0;
             XChangeProperty(x11_display, x11_window, overlay_atom, XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&value, 1);
             XSync(x11_display, 0);
